language/tools/move-mv-llvm-compiler/tests/test_common.rs

```diff
@@ -790,3 +792,116 @@ pub fn clean_results<P: AsRef<Path>>(source: P) -> std::io::Result<()> {
 
     Ok(())
 }
+
+pub fn switch_last_two_extensions_and_rename(path: &Path) {
+    if let Some(filename) = path.file_name() {
+        if let Some(filename_str) = filename.to_str() {
+            let mut parts: Vec<&str> = filename_str.split('.').collect();
+
+            if parts.len() >= 3 {
+                // Swap the last two extensions
+                let last_ext = parts.pop().unwrap();
+                let second_last_ext = parts.pop().unwrap();
+                parts.push(last_ext);
+                parts.push(second_last_ext);
+
+                let new_filename = parts.join(".");
+
+                let mut new_path = path.to_path_buf();
+                new_path.set_file_name(new_filename);
+
+                fs::rename(path, new_path).expect("Error in renameing");
+            }
+        }
+    }
+}
+
+pub fn list_files_with_extension(
+    directory_path: &Path,
+    extension: &str,
+) -> std::io::Result<Vec<String>> {
```

Comment:
```suggestion
) -> std::io::Result<Vec<PathBuf>> {
```

Reply:
Why pass Strings around if you always convert them back to PathBuf in every use of list_files_with_extension results?